divide_and_conquer/fibonacci_sequence_dnc.py

```diff
@@ -0,0 +1,63 @@
+"""
+We can find the Fibonacci sequence more effectively than naive bottom-up algorithm
+    We use One of the well-known properties of Fibonacci number is:
+
+    | F(n+1) ,  F(n)   |       | 1 , 1 | ^ n
+    |                  |   =   |       |
+    | F(n)   ,  F(n-1) |       | 1 , 0 |
+
+    We can therefore compute the (n)th power of A = ((1,1),(1,0)) using recursive squaring
+
+    Time complexity: O(log(n)), because it is a divide and conquer algorithm
+"""
+
+
+def get_fibonacci(n):
+    """
+    :param n: number of Sequence
+    :return: (n)th sequence of Fibonacci sequence
+
+    >>> get_fibonacci(5)
+    5
+    >>> get_fibonacci(10)
+    55
+    >>> get_fibonacci(20)
+    6765
+    """
+    if n < 2:
+        matrix = [[1, n], [1, 0]]
+        return matrix[0][1]
+    else:
+        matrix = [[1, 1], [1, 0]]
+        return get_squared(matrix, n)[0][1]
+
+
+def get_squared(matrix, n):
```

Comment:
I am sorry if i was unclear, but you need to write **doctests** for all the _methods/functions_, taking into considerations all the cases i.e covering all the branches of your code.

Check this out for idea https://github.com/TheAlgorithms/Python/blob/master/data_structures/linked_list/circular_linked_list.py